smartapps/bangali/rooms-child-app.src/rooms-child-app.groovy

```diff
@@ -2410,27 +2413,165 @@ def processCoolHeat()       {
 
 def processCoolHeat()       {
```

Comment:
Split into three methods, setTemp() which does the actual heating/cooling, processCoolHeat for the logic and getActiveTemperatureRule() which is unchanged otherwise.